From 58fd919590fbc52db354c1a3e74683f83d183b0d Mon Sep 17 00:00:00 2001
From: Dhaval Khandla <dhavaljk@ti.com>
Date: Fri, 31 Jan 2020 18:04:03 +0530
Subject: [PATCH 5/6] HACK: arm64: dts: ti: Disable SMMU for PCIe

For some applications, the region exposed over PCIe might not be
allocated by Linux. There is no easy way to program that region in SMMU.
Hence, enabling SMMU creates faults on the host connected to NTB. So it
needs to be disabled.

Signed-off-by: Dhaval Khandla <dhavaljk@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-main.dtsi | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
index 8260d323c164..70745bd614d2 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
@@ -1001,7 +1001,6 @@
 			vendor-id = /bits/ 16 <0x104c>;
 			device-id = /bits/ 16 <0xb00d>;
 			msi-map = <0x0 &gic_its 0x0 0x10000>;
-			iommu-map = <0x0 &smmu0 0x0 0x1000>;
 			dma-coherent;
 			ranges = <0x01000000 0x0 0x10001000  0x00 0x10001000  0x0 0x0010000>,
 				 <0x02000000 0x0 0x00000000  0x40 0x00000000  0x0 0x80000000>;
@@ -1049,7 +1048,6 @@
 			      <0x00 0x18000000 0x00 0x00001000>;
 			reg-names = "reg", "cfg";
 			msi-map = <0x0 &gic_its 0x10000 0x10000>;
-			iommu-map = <0x0 &smmu0 0x04000 0x1000>;
 			dma-coherent;
 			ranges = <0x01000000 0x0 0x18001000  0x00 0x18001000  0x0 0x0010000>,
 				 <0x02000000 0x0 0x00000000  0x41 0x00000000  0x0 0x80000000>;
@@ -1097,7 +1095,6 @@
 			      <0x44 0x00000000 0x00 0x00001000>;
 			reg-names = "reg", "cfg";
 			msi-map = <0x0 &gic_its 0x10000 0x10000>;
-			iommu-map = <0x0 &smmu0 0x08000 0x1000>;
 			dma-coherent;
 			ranges = <0x01000000 0x00 0x00001000  0x44 0x00001000  0x0 0x0010000>,
 				 <0x02000000 0x00 0x00011000  0x44 0x00011000  0x0 0x7fef000>;
-- 
2.17.1

